modules/mediaforceBidAdapter.js

```diff
@@ -140,31 +150,47 @@ export const spec = {
           default: return;
         }
       }
-      validImp && imp.push(impObj);
 
-      let request = {
-        id: bid.transactionId,
-        site: {
-          page: referer,
-          ref: referer,
-          id: bid.params.publisher_id,
-          publisher: {
-            id: bid.params.publisher_id
+      let request = requestsMap[bid.params.publisher_id];
+      if (!request) {
+        request = {
+          id: Math.round(Math.random() * 1e16).toString(16),
+          site: {
+            page: referer,
+            ref: referer,
+            id: bid.params.publisher_id,
+            publisher: {
+              id: bid.params.publisher_id
+            },
           },
-        },
-        device: {
-          ua: navigator.userAgent,
-          js: 1,
-          dnt: dnt,
-          language: getLanguage()
-        },
-        imp
-      };
-      requests.push({
-        method: 'POST',
-        url: bid.params.is_test ? TEST_ENDPOINT_URL : ENDPOINT_URL,
-        data: JSON.stringify(request)
-      });
+          device: {
+            ua: navigator.userAgent,
+            js: 1,
+            dnt: dnt,
+            language: getLanguage()
+          },
+          ext: {
+            mediaforce: {
+              hb_key: auctionId
+            }
+          },
+          tmax: timeout,
+          imp: []
+        };
+        requestsMap[bid.params.publisher_id] = request;
+        requests.push({
+          method: 'POST',
+          url: ENDPOINT_URL,
+          data: request
+        });
+      }
+      validImp && request.imp.push(impObj);
+    });
+    requests.forEach((req) => {
+      if (isTest) {
+        req.url = TEST_ENDPOINT_URL;
+      }
+      req.data = JSON.stringify(req.data);
```

Comment:
Any reason for the separate loop, couldn't this be applied when building the request above?

Reply:
The parameter "isTest" can be configured in any bid in the middle of list, and therefore we need make second loop after all bid was processed. And some bids in theory can have different publisher_id in params, and because the publisher_id defines site field, which is common to all imps, we must prepare separate request for each publisher_id.

Reply:
Thanks, I suppose you expect to have different publisher_id values with different site values from the same publisher/site. I don't see why that would occur but I'm not against your adapter supporting it.

For isTest, wouldn't you want this to be configured on a per request basis? For example, publisher_id: 203 want's test debug request url, but publisher_id: 204 in a different request does not. In this configuration, publisher_id: 204 would use the test endpoint. Again, fine with this but would like to confirm.

Reply:
isTest parameter used in the main in case to test new changes before commit it. And therefore i would to leave it as is.